### PR TITLE
[Vue] migrate the plugin-settings directive

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of #18431.**

Changes:
* Migrate the plugin-settings directive.
* Allow FormField.vue to execute conditions without supplying all-settings param. Doesn't make sense in a Vue context since the parent component can keep track of the list of all values. Still there, but only for BC.
* Add AjaxHelper.post() method.

**IMPORTANT NOTE:** This PR is still using angularjs for expression evaluation. There doesn't seem to be an established library to do something similar, though a handful exist (none of them seem very popular). We'll either have to use one of those or roll our own. I'm not sure which is the best option, but suggestions are welcome.

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
UI tests passing for this one (some screenshots in plugins need an update, but holding off on that until after the specific PRs are merged)

File: plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue
Comment (peterhashair) [inline plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue:194]: just a question. do we want to implement a test for the AjaxHelper, or is actually a part of the integration UI test?

**Comment by peterhashair** [comment]
@diosmosis hasn't finished reviewing the VUE code on that one, but did resolve conflicts for that one, can you review the merge conflicts, in case I merge the wrong parts.

File: package.json
Comment (sgiehl) [inline package.json:81]: Is that one only needed for building the matomo polyfill? Guess it was on purpose that it's code wasn't added to `node_modules`?

File: plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue
Comment (sgiehl) [inline plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue:112]: The part that was added in https://github.com/matomo-org/matomo/pull/18404/files#diff-3fe2ecdd62c8a5f38ede22ec60896debbfe870bdde10b889f78d129fc5a3ffa2 got lost

File: plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue
Comment (sgiehl) [inline plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue]: There is kind of a bug around the `passwordConfirmation`. Wondering if it would make sense to directly apply a fix here, instead of later adjusting https://github.com/matomo-org/matomo/pull/18453

File: plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue
Comment (sgiehl) [inline plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue:67]: Wondering if adding that might make sense, as otherwise the page reloads when clicking on "no"
```suggestion
          class="modal-action modal-close modal-no"
          @click="$event.preventDefault();"
```
Or maybe that needs to be fixed in the modal, as it wasn't the case before 🤔

File: plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue
Comment (sgiehl) [inline plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue]: If that's useful and/or needed, might be good to create an follow up issue on github instead of a code comment. Otherwise I guess this will be never addressed 🙈

**Review by sgiehl — CHANGES_REQUESTED** [review]
Generally seems to work. Found a couple of smaller issues that needs to be fixed.

File: package.json
Comment (diosmosis) [reply package.json:81]: Yes, the contents will be included in MatomoPolyfills.umd.min.js, so it's not needed to include the node_module as well in git. Only the libraries that are included directly in Matomo's asset pipeline have to live in git.

File: plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue
Comment (diosmosis) [reply plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue]: If the other one is ready to merge, it could be merged and I can then convert it to Vue here?

Comment (peterhashair) [reply plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue:153]: should I pull this PR and make a change on here?

**Comment by peterhashair** [comment]
@diosmosis did a fix for one of the issues in #18453 for this version here. When we type a wrong password in the modal, then close it, `passwordConfirmation` still has the value. Separate the `modal-show` and `save` it into 2 different functions.
Other than that, it's working as expected.

**Comment by sgiehl** [comment]
@diosmosis would be good to review #18515 and merge it before pushing any further changes, to avoid merge conflicts.

File: plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue
Comment (diosmosis) [reply plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSettings.vue:194]: Adding new tests is out of scope for the Vue migration (for me at least), I'm sure it's possible to add them some day in the future however. Perhaps creating a new issue would be a good idea.

**Comment by diosmosis** [comment]
@sgiehl I will take a look as soon as I can but I will be pretty busy with other things for the moment.

**Comment by sgiehl** [comment]
@diosmosis no problem. Btw. I found another regression in TagManager (top controls aren't visible anymore). Shall I create a issue for that and assign it to you?

**Comment by diosmosis** [comment]
@sgiehl :+1: sure, go ahead!

File: plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSetting.vue
Comment (sgiehl) [inline plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSetting.vue]: Opening the settings page logs a couple of warnings in the browser console:
image
Not sure if that's something that needs to be fixed or not.

Comment (diosmosis) [reply plugins/CorePluginsAdmin/vue/src/PluginSettings/PluginSetting.vue]: 👍 worth fixing

**Review by sgiehl — APPROVED** [review]

**Pull request merged**
